test/e2e/functional/tests/e2e.go

```diff
@@ -163,6 +163,9 @@ var _ = Describe("metallb", func() {
 
 			Expect(testclient.Client.Create(context.Background(), addresspool)).Should(Succeed())
 
+			defer (func() {
```

Comment:
I'd try to avoid deferring the cleanup, and having a more organic way to delete the address pools (or even all the CRDs) in AfterEach.

A reusable functoin that iterates over all the CRs (pools, peers, bfd) and delete them would be probably useful to ensure state do not leak between tests.
With that, we can add it to the outer AfterEach without the need of keeping a reference to what's created, because what we really care about is having a clear slate when the next test starts.

Reply:
don't we want to have the option to run the e2e-test on clusters with already deployed metallb?
(i.e - metallbCRExisted = true)
in this case, how will I be able to know what CRs are part of the test and what belongs to the user?
(without keeping a reference to what's created)

Reply:
Not at the moment, at least. If we want to keep that option, we could mark the resources with a unique prefix so we can recognize them.